test/Interop/Cxx/value-witness-table/basic.swift

```diff
@@ -0,0 +1,134 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop)
+//
+// REQUIRES: executable_test
+
+import Basic
+import StdlibUnittest
+
+var CXXDestructorTestSuite = TestSuite("CXXDestructor")
+
+protocol InitWithPtr {
+  init(value: UnsafeMutablePointer<Int32>!)
+}
+
+extension HasUserProvidedDestructor : InitWithPtr { }
+
+protocol InitWithBox {
+  init(box: HasUserProvidedDestructor)
+}
+
+extension EmptyDestructor : InitWithBox { }
+
+@inline(never)
+func createTypeWithUserProvidedDestructor(_ ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+}
+
+@inline(never)
+func createTypeWithEmptyDestructor(_ ptr: UnsafeMutablePointer<Int32>) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let empty = EmptyDestructor(box: box)
+}
+
+@inline(never)
+func createTypeWithNonTrivialImplicitDestructor(
+  _ ptr: UnsafeMutablePointer<Int32>
+) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let implicit = HasNonTrivialImplicitDestructor(box: box)
+}
+
+@inline(never)
+func createTypeWithNonTrivialDefaultDestructor(
+  _ ptr: UnsafeMutablePointer<Int32>
+) {
+  var box = HasUserProvidedDestructor(value: ptr)
+  let def = HasNonTrivialDefaultDestructor(box: box)
+}
+
+@inline(never)
+func createTypeWithGeneric<T : InitWithPtr>(
+  _ ptr: UnsafeMutablePointer<Int32>,
+  type : T.Type
+) {
+  let box = T(value: ptr)
+}
+
+@inline(never)
+func createTypeWithProtocol(
+  _ ptr: UnsafeMutablePointer<Int32>,
+  type : InitWithPtr.Type
```

Comment:
```suggestion
  type: InitWithPtr.Type
```

Please apply throughout the PR.